_includes/js/files.md

```diff
@@ -113,20 +113,19 @@ Parse.Cloud.httpRequest({ url: profilePhoto.url() }).then(function(response) {
 
 ## Deleting Files
 
-You can delete files that are referenced by objects using the [REST API]({{ site.baseUrl }}/rest/guide/#deleting-files). You will need to provide the master key in order to be allowed to delete a file.
-
-If your files are not referenced by any object in your app, it is not possible to delete them through the REST API.
-
-Alternatively, starting with Parse Server 4.2.0, you can delete files using cloud code.
+You can delete files that are referenced by objects using the `destroy` method. The master key is required to delete a file.
 
 ```javascript
-Parse.Cloud.beforeDelete('Profile', async (req) => {
-  const profile = req.object;
-  const profilePhoto = profile.get("photoFile");
-  await profilePhoto.destroy({ useMasterKey: true })
-});
+const profilePhoto = profile.get("photoFile");
+await profilePhoto.destroy({ useMasterKey: true });
 ```
 
+If your file is not referenced by any object in your app, it is not possible to delete it through Parse.
```

Comment:
I believe it is actually possible to delete files not referenced by objects if you know the file name (same requirement for deleting using the REST API):

```js
(new Parse.File('fileName')).destroy();
```

Reply:
Oh, that is interesting. I wonder if that should be a documented feature or is it more of a hack (calling destroy on an unsaved file object)? If it's supposed to be a feature, then we'd need to have a test case for this.

Reply:
I looked into Cloud Code test cases and it seems that this pattern of creating a file (without saving it) and then deleting it is used many times.

The REST API docs say "send a DELETE request to the files URL, postfixed with the name of the file." which is similar to creating a `ParseFile` object, setting a name (without saving it) and then calling destroy on it. While it has a "hacky" appearance, the `Parse.File.destroy()` method is doing the same as the REST API call.

However, there are some potential issues:

- The `Parse.File.destroy` method shows that the terms "saving" and "naming" a file are used interchangeably, which could lead to confusion:

```
if (!this._name) {
  throw new Error('Cannot delete an unsaved ParseFile.');
}
```

- There is no internal ParseFile state parameter that indicates whether the file has been saved or only named, but there is a change of the file name after saving a file, as the [JS SDK API reference](http://parseplatform.org/Parse-SDK-JS/api/2.18.0/Parse.File.html#name) says:
> Before save is called, this is the filename given by the user. After save is called, that name gets prefixed with a unique identifier.

That concludes that "naming" a file is not the same as "saving" a file, because "saving" actually changes the file name. Therefore I would suggest to not explicitly describe it as an official feature that a `ParseFile` object can be named and then destroyed, without ever being saving. The officially supported and "clean" process would be to create a file, keep a reference to it, and delete it by that given reference.

Reply:
I believe we should change the error message `Cannot delete an unsaved ParseFile` and I'm ok in not mentioning it can be done but I'd also not state that it can not be done.

Reply:
Sounds good, let me remove that line.